school_library.rb

```diff
@@ -11,12 +13,44 @@ def initialize(age, name = 'Unknown', parent_permission: true)
 
   attr_accessor :age
 
-  Private def of_age
-    @age >= 18
-  end
   def can_use_services
     return unless @age >= 18 || parent_permission
 
     true
   end
+
+  def correct_name
+    @name
+  end
+
+  private
+
+  def of_age
+    @age >= 18
+  end
+end
+
+class BaseDecorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.upcase!
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.splice(0, 9)
```

Comment:
- [ ] This returns an error. You should call the `correct_name` method as well.

- [ ] There is no method called `splice` in Ruby, I think you meant `#slice`. Nonetheless, I think we can save some code here.

  ```
  @nameable.correct_name[0, 10]
  ```